### PR TITLE
Wip serialize iou object

### PR DESCRIPTION
**Pull request by joshwalong**

### CODE REVIEW

File: binary-codec/types/type_utils.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+			if decimalSuffixNoTrailingZeros == "" { // if the decimal suffix is empty because it only contained trailing zeros
+				bigDecimal.Scale = len(decimalPrefixNoLeadingZeros) - len(bigDecimal.UnscaledValue) // set the scale to the difference between the length of the decimal prefix and the length of the unscaled value
+			}
+
+		} else if !containsDecimal && decimalErr == nil { // if the value does not contain a SINGLE decimal point
+			decimalPrefixNoTrailingZeros := strings.TrimRight(prefix, "0")             // remove trailing zeros from the prefix
+			bigDecimal.Scale = len(prefix) - len(decimalPrefixNoTrailingZeros)         // set the scale to the difference between the length of the prefix and the length of the prefix with trailing zeros removed
+			bigDecimal.UnscaledValue = strings.Trim(decimalPrefixNoTrailingZeros, "0") // remove leading and trailing zeros from the prefix with trailing zeros removed to get the unscaled value
+		} else {
+			return nil, err // Do I need this?
```
Comment (bmurphy333) [inline binary-codec/types/type_utils.go]: do you need this?

File: binary-codec/types/type_utils.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+type InvalidCodeError struct {
+	Disallowed string
+}
+
+func (e *InvalidCodeError) Error() string {
+	return fmt.Sprintf("'%s' is/are disallowed or invalid", e.Disallowed)
+}
+
+// Creates a new custom BigDecimal object from a value string
+func NewBigDecimal(value string) (*BigDecimal, error) {
```
Comment (bmurphy333) [inline binary-codec/types/type_utils.go]: lets move this and the struct into a pkg/bigDecimal package. Cause can get used outside of binary codec

File: pkg/big-decimal/big_decimal.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	"strconv"
+	"strings"
+)
+
+const (
+	AllowedCharacters = "0123456789.-eE"
+	BigDecRegEx       = "-?(?:[0|1-9]\\d*)(?:\\.\\d+)?(?:[eE][+\\-]?\\d+)?"
+)
+
+var ErrInvalidCharacter = fmt.Errorf("value contains invalid characters. Only the following are allowed: %q", AllowedCharacters)
```
Comment (JCrawsh) [inline pkg/big-decimal/big_decimal.go]: can we have grouped vars?

**Review by JCrawsh — APPROVED** [review]
LGTM

**Pull request merged**